### ISSUE TITLE
Eagerly connect CLI to server when --external-authentication option is used

### ISSUE THREAD

**Issue by findepi** [trinodb/trino#9441]
When CLI is used with password- or kerberos-based auth, it doesn't need to actually make a connection on startup. 
CLI prints the prompt and authentication happens behind the scenes.

When `--external-authentication`, the user didn't provide authentication information yet. Printing prompt, and authenticating on first query isn't the best user experience possible.

**Comment by kokosing** [comment]
I think it is not only about external authentication. For example one would like to know that the password (or keytab) is invalid eagerly too.

**Comment by kokosing** [comment]
Oh... that would also verify if the server is up and running too.

**Comment by findepi** [comment]
@kokosing  I agree that it would be better UX overall, but i also am aware the way CLI works was a conscious decision (even if i would make a different one).
That's why I would want to focus on `--external-authentication` case only.

**Comment by electrum** [comment]
Yes, we should change this for everything, not just authentication.
